__tests__/ExpensiMark-HTML-test.js

```diff
@@ -985,6 +985,26 @@ test('Test for @here mention without space or supported styling character', () =
     expect(parser.replace(testString)).toBe(resultString);
 });
 
+test('Test for mention inside link and wrapped by styles like italic, bold, strikethrough or code', () => {
```

Comment:
Can you add tests for email as well e.g. `[_@username@expensify.com_](username@expensify.com)`

Also we have `Test for user mention inside link markdown` and `Test for user mention inside email markdown` tests, maybe we can merge those here.